up.sh

```diff
@@ -41,94 +50,86 @@ fi
 
 git reset --hard
 git checkout master
-git reset --hard upstream/master
+git reset --hard origin/master
 
-DERIVATION_FILE=$(find . | grep "/$1/default.nix" | head -n1) || error_exit "Couldn't find derivation file."
+DERIVATION_FILE=$(nix-instantiate --eval -E "with import ./. {}; (builtins.unsafeGetAttrPos \"name\" pkgs.$PACKAGE_NAME).file" | sed 's/^"\(.*\)"$/\1/' || echo "")
```

Comment:
I think this is a more robust way to find filename. Do you think this will work as a solution? You can cherry-pick this part as it's most useful.

Reply:
Actually it needs to be tweaked a little bit.